instance/serializers/instance.py

```diff
@@ -110,3 +114,21 @@ class InstanceReferenceDetailedSerializer(InstanceReferenceBasicSerializer):
     more detail.
     """
     summary_only = False
+
+
+class InstanceAppServerSerializer(serializers.ModelSerializer):
+    """
+    Provide the complete list of app servers for an instance
+    """
+    class Meta:
+        model = InstanceReference
+        fields = ('app_servers',)
+
+    def to_representation(self, obj):
+        output = super().to_representation(obj)
+        request = self.context['request']
+        output['app_servers'] = [
+            AppServerBasicSerializer(appserver, context={'request': request}).data
```

Comment:
I think you should just say `context=self.context` instead of declaring a local `request` variable and passing it in explicitly like this.

Reply:
Done